Add --version flag to CLI

Dahlializi opened the pull request:
Summary
- Add `--version` to the `cautious-robot` CLI so users can check the installed version without relying on `pip list`.

Implementation
- Follow the same pattern used in TaxonoPy: expose a package-level `__version__` and use argparse's built-in `action="version"`.

Testing
- `(cautious-robot-env) [dahlia@cardinal-login01 cautious-robot]$ cautious-robot --version`
`cautious-robot 1.1.0`

Commit:
Add --version flag to CLI

hlapp reviewed (approved):
👍🏻

thompsonmj reviewed (changes requested):
@egrace479 raises a good point. I think the better option is `__about__.py`. @Dahlializi, could you update this PR accordingly?

Commit:
Use __about__.py as single source of version for CLI

Dahlializi commented:
@thompsonmj Thanks for the clarification, that makes sense!
I’ve updated the PR to use `__about__.py` as the single source of truth for the version, and adjusted the CLI to read `__version__` from there instead of `__init__.py`.

hlapp reviewed (approved):
👍🏻

thompsonmj reviewed (approved):
Looks good!
